tools/scripts/rebuildAnimationLibraryManifest.rb

```diff
@@ -0,0 +1,205 @@
+    alias_map = {}
+    animation_metadata_by_name.each do |name, metadata|
+      aliases = [name] + (metadata['aliases'] || [])
+      aliases.each do |aliaz|
+        # Push name into target array, deduplicate, and sort
+        alias_map[aliaz] = ((alias_map[aliaz] || []) + [name]).uniq.sort
+      end
+      alias_progress_bar.increment unless @options[:quiet]
+    end
+    alias_progress_bar.finish unless @options[:quiet]
+
+    alias_map.each {|k, v| verbose "#{bold k}: #{v.join(', ')}"} if @options[:verbose]
+
+    info "Mapped #{alias_map.size} aliases."
+
+    # Write result to file
+    File.open(DEFAULT_OUTPUT_FILE, 'w') do |file|
+      file.write(JSON.pretty_generate({
+          '//': [
+              'Animation Library Manifest',
+              'GENERATED FILE: DO NOT MODIFY DIRECTLY',
+              'See tools/scripts/rebuildAnimationLibraryManifest.rb for more information.'
+          ],
+          'metadata': animation_metadata_by_name,
+          'aliases': alias_map
+      }))
+    end
+    info <<-EOS.unindent
+      Manifest written to #{DEFAULT_OUTPUT_FILE}.
+
+        #{dim 'd[ o_0 ]b'}
+    EOS
+
+  # Report any issues while talking to S3 and suggest most likely steps for fixing it.
+  rescue Aws::Errors::ServiceError => service_error
+    warn service_error.inspect
+    warn <<-EOS.unindent
+
+      #{bold 'There was an error talking to S3.'}  Make sure you have credentials set using one of:
+
+        * aws_access_key and aws_secret_key in your locals.yml
+        * ENV['AWS_ACCESS_KEY_ID'] and ENV['AWS_SECRET_ACCESS_KEY']
+        * ~/.aws/credentials
+
+      #{dim 'See http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html for more details.'}
+    EOS
+  end
+
+  def verbose(s)
+    puts(s) if @options[:verbose]
+  end
+
+  def info(s)
+    puts(s) unless @options[:quiet]
+  end
+
+  def warn(s)
+    puts(s)
+  end
+end
+
+# Given the complete body of a PNG file, returns hash of source dimensions:
+# {"x": _, "y": _}
+def dimensions_from_png(png_body)
+  # Read the first eight bytes of the IHDR Chunk, which must always be the
+  # first chunk of the PNG file.
+  #
+  # PNG Header takes 8 bytes (0x00-0x07)
+  # IHDR chunk length takes 4 bytes (0x08-0x0b)
+  # IHDR chunk type code takes 4 bytes (0x0c-0x0f)
+  # IHDR chunk data begins at 0x10
+  #
+  # The IHDR chunk begins with
+  # 4 bytes for width (0x10-0x13) followed by
+  # 4 bytes for height (0x14-0x17)
+  #
+  # PNG File Structure
+  # http://www.libpng.org/pub/png/spec/1.2/PNG-Structure.html
+  # IHDR Chunk Layout
+  # http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.IHDR
+  dimensions = png_body[0x10..0x18].unpack('NN')
+  {'x': dimensions[0], 'y': dimensions[1]}
+end
+
+# Parse command-line options and then start the rebuild process
+options = {}
+cli_parser = OptionParser.new do |opts|
+  opts.banner = "Usage: ./rebuildAnimationLibraryManifest.rb [options]"
+  opts.separator ""
+  opts.separator "Options:"
```

Comment:
Lifted from `bin/dotd` so here's my best understanding:  One of `OptionParser`'s features is to generate an "option summary" (read: help text) from the options you define in the order you define them.  The `separator` method lets you inject text into that summary, wherever you call it relative to where the options are defined.  Here we are inserting a blank line and an "Options:" line.  The result looks like this:

```
╰○ ./tools/scripts/rebuildAnimationLibraryManifest.rb --help
Usage: ./rebuildAnimationLibraryManifest.rb [options]

Options:
    -q, --quiet                      Only log warnings and errors
    -v, --verbose                    Use verbose log output
    -h, --help                       Show this message
                                                      
```